### DISCUSSION TITLE
stress vs virial weights

### DISCUSSION THREAD

**Discussion by gabor1** [ACEsuit/mace#521] [General]
when we specify stress or virial, one of these presumably gets converted before being used in the loss. what happens to the loss weight? the user expectation might be that if I specify stress, then the weight is intensive, and that if I specify virial, then it's extensive. in GAP we specifically stated that the loss weights for virials are really virial/atom (same as for energy), so both are intensive in fact. what's the situation in MACE ?

**Comment by ilyes319** [comment]
At the moment, the virials and the stress are used separately and not converted internally, which is a bad thing, and we should change that. We do divide the virial loss by the number of atoms, so the weight one provides is "per atom".

**Comment by gabor1** [comment]
so what happens when some data has stress and others have virial? I guess you need to supply loss weights for both?

**Reply by ilyes319** [reply]
We do not support mix data with stress and virial at the moment. By default, if virials are not enforced, it will train on stresses. I think we should unify the losses to only use the stress internally if this makes the most sense. I know sometime one can define the virials and not the stress, so I am not sure if it is the best.

**Comment by gabor1** [comment]
ok. let's create an issue and assign it to Eszter :-)
